### DISCUSSION TITLE
Adding Push Notification to VirtualKeypad-Web

### DISCUSSION THREAD

**Discussion by hcoiro** [taligentx/dscKeybusInterface#215] [dscKeybusInterface Discussion]
I am working on adding push notification to the VirtualKeypad-Web. 
I am using the site  https://pushme.now.sh  to send notificationthat also have a code in github to be used and mount your own push notification site.
I am basically using VirtualKeypad-Web code, adding some code from EMAIL  and PUSHBULLET for notification.
Also I did use another WIFI library, that when not have wifi network configured, you can introduce it accessing via web to the ESP8266 and congigure the AP and password.
Also changing the web front end to allow you to enter the code that give you the site pushme.now.sh for 6 different notifications web browser of 6 user.
I can share the code with anyone, and also can share here to be posted. 
Have nothing of experience in github and this collaborative tools, so if someone is interested, I can send the file, what I don´t know if there is some contition inside the code to comply in order to share it. Please let me know if you would like to have the code.

**Comment by qighu21** [comment]
Hello, I am using Home Assistant - MQTT example for my DSC 1616, it works great. Now I am working to add email notification to it, but I am having problem to connect ESP8266 to the email server. Could you send me your code via email so I may be able to use some of your code for my email notification function. my email address is qighu@hotmail.com
Thank you,
William

**Reply by Edissonm60** [reply]
amigo me podria prestar el codigo edissonm60@gmail.com gracias

**Reply by Fabrizz** [reply]
> Hello, I am using Home Assistant - MQTT example for my DSC 1616, it works great. Now I am working to add email notification to it, but I am having problem to connect ESP8266 to the email server. Could you send me your code via email so I may be able to use some of your code for my email notification function. my email address is qighu@hotmail.com
> Thank you,
> William

Why do you want to use the ESP for the notifications? Is better and easier to use HomeAssistant as the handler and coordinator for the email notifications

**Reply by qighu21** [reply]
I am new to Home Assistant, don't know how to setup email/SMS notifications. but I have successfully added email & SMS to ESP8266 sketch.

**Reply by Edissonm60** [reply]
gighu 21 amigo tienes el codigo para correo y smses que no me funciona el que aparece y trato de unir dos codigos y no me dejs gracias

**Comment by Dilbert66** [comment]
If you are interested, I do have a version designed for the Honeywell Vista that is partly based on the Virtualkeypad example here. It uses Telegram for push notifications and control. I  used Telegram as my preferred notification app since it provides the use of bots which allows full control of the alarm system using the telegram client remotely as well as push notifications on events at no cost.

You just need the telegram_async.h and telegram_async.cpp files to add the function to your app.  You will then need to configure the pushlib with your userid, bot token  then add the notifications call wherever you want the push notification to go out.

Please note that it's designed for the ESP32 only.  The ESP8266 was too low on resources to run all the components as well as lacking true async multitasking.  Having the telegram client running on its own task allows it run as a bot and will not impact the alarm panel control code.

https://github.com/Dilbert66/VirtualKeypad-Vista
